pydash-front/src/login/Login.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import { Route, Redirect } from 'react-router'
+import './Login.css';
+import Button from 'material-ui/Button';
+import TextField from 'material-ui/TextField';
+import axios from 'axios';
+
+class Login extends Component {
+    state = {
+        username: '',
+        password: '',
+        error: false,
+        message: '',
+        success: false
+    };
+
+    handleChange = key => event => {
+        this.setState({
+            [key]: event.target.value
+        });
+    };
+
+    tryLogin = (e) => {
+        e.preventDefault()
+        let username = this.state.username,
+            password = this.state.password
+        
+        if (!(username.trim()) || !(password.trim())) {
+            return;
+        }
+
+        // Make a request for a user with a given ID
+        axios.post('http://localhost:5000/api/login/', {
+            username,
+            password
+        }).then((response) => {
+            console.log(response);
+            this.setState(prevState => ({
+                error: false,
+                helperText: '',
+                success: true
+            }))
+        }).catch((error) => {
+            console.log(error);
+            this.setState(prevState => ({
+                error: true,
+                helperText: 'Incorrect credentials 😱'
+            }))
+        });
+    }
+
+    render() {
+        return this.state.success ? (
```

Comment:
might be better to use if ... else ...

Reply:
In the React.js router it's quite normal to use compound expressions: [https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Redirect.md](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Redirect.md)